html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -86,22 +90,36 @@ public function add()
 		$biditem = $this->Biditems->newEntity();
 		// POST送信時の処理
 		if ($this->request->is('post')) {
+			//ファイル情報を変数に入れておく
+			$fileinfo = $this->request->getData('image_path');
 			// $biditemにフォームの送信内容を反映
 			$biditem = $this->Biditems->patchEntity($biditem, $this->request->getData());
-			// $biditemを保存する
-			if ($this->Biditems->save($biditem)) {
-				// 成功時のメッセージ
-				$this->Flash->success(__('保存しました。'));
-				// トップページ（index）に移動
-				return $this->redirect(['action' => 'index']);
+			//バリデーションのチェック　エラーがあればエラー内容を表示し再入力画面へ
+			if ($biditem->errors()) {
+				$this->Flash->error(__(''));
+			} else {
+				//一時的に'image_path'に仮の値を入れておく
+				$biditem['image_path'] = "temporaryimagepath";
+				if ($this->Biditems->save($biditem)) {
+					//登録した商品IDを取得
+					$biditem_id = $biditem->id;
+					//imageパスを作成
+					$imagepath = '../webroot/img/auction/' . $biditem_id . '.' . pathinfo($fileinfo['name'], PATHINFO_EXTENSION);
+					move_uploaded_file($fileinfo['tmp_name'], $imagepath);
+					//再度'image_path'カラムに値を商品ID＋拡張子で保存
+					$biditem['image_path'] = $biditem_id . '.' . pathinfo($fileinfo['name'], PATHINFO_EXTENSION);
```

Comment:
$biditem_id . '.' . pathinfo($fileinfo['name'], PATHINFO_EXTENSION)

は変数に入れて使った方が安全ですね。

```suggestion
					$newFileName = $biditem_id . '.' . pathinfo($fileinfo['name'], PATHINFO_EXTENSION);
					$imagepath = '../webroot/img/auction/' . $newFileName;
					move_uploaded_file($fileinfo['tmp_name'], $imagepath);
					//再度'image_path'カラムに値を商品ID＋拡張子で保存
					$biditem['image_path'] = $newFileName;
```